obspy/io/sac/tests/test_sacpz.py

```diff
@@ -69,9 +69,9 @@ def test_write_sacpz_soh(self):
                             'data', 'only_soh.xml')
         inv = read_inventory(path)
         f = io.StringIO()
-        with warnings.catch_warnings(record=True) as w:
-            inv.write(f, format='SACPZ')
-            self.assertEqual(len(w), 2)
+        inv.write(f, format='SACPZ')
+        # Only 2 newlines are written.
+        self.assertEqual(2, f.tell())
```

Comment:
What is wrong with testing the warning? Hhow you were doing it was fragile though.. check out other parts of the code base that test for warning messages (should ideally be refactored in a utility routine at some point..)

```python
+        # check emitted warning message
+        self.assertTrue(len(w) >= 1)
+        msg = ("'components' is expected to be a list of one or more "
+               "component code combinations (e.g. `components=['Z12']` or "
+               "`components=['Z12', '123']`)")
+        for w_ in w:
+            try:
+                self.assertEqual(str(w_.message), msg)
+            except AssertionError:
+                continue
+            else:
+                break
+        else:
+            self.fail(msg='Expected warning message not shown.')
```